components/builder-scheduler/src/server/logger.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2016-2017 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fs::File;
+use std::io::Write;
+use std::path::PathBuf;
+use chrono::prelude::*;
+
+use protocol::jobsrv::Job;
+use protocol::scheduler::{Group, Project};
+
+pub struct Logger {
+    file: File,
+}
+
+impl Logger {
+    pub fn init(log_path: PathBuf) -> Self {
```

Comment:
Whenever we have a function that takes a `Path` or a `PathBuf` we should always use a signature like this:

```
pub fn init<T>(log_path: T> -> Self where T: AsRef<Path> {
    log_path.as_ref().to_path_buf()
}